shoop/campaigns/admin_module/form_sets.py

```diff
@@ -58,11 +58,24 @@ def get_queryset(self):
         return self.owner.conditions.instance_of(self._get_actual_model())
 
 
-class BasketEffectsFormSet(BaseFormset):
+class BasketDiscountEffectsFormSet(BaseFormset):
     model = BasketDiscountEffect
 
     def get_queryset(self):
-        return self.owner.effects.instance_of(self._get_actual_model())
+        return self.owner.basket_discount_effects.instance_of(self._get_actual_model())
+
+    def form(self, **kwargs):
+        if issubclass(self.form_class, TranslatableModelForm):
+            kwargs.setdefault("languages", settings.LANGUAGES)
+            kwargs.setdefault("default_language", settings.PARLER_DEFAULT_LANGUAGE_CODE)
+        return self.form_class(initial={"campaign": self.owner}, **kwargs)
```

Comment:
Actually.. I see now that it's copypasted from the other formset. Why don't you just provide an abstract base class to DRY this up. Define `form` method in the abstract base and `model` and `get_queryset` in the inherited classes:

```
class EffectsFormset(BaseFormset):
    def form(self, **kwargs):
        kwargs.setdefault("initial", {"campaign": self.owner})
        return super(EffectsFormset, self).form_class(**kwargs)

class BasketDiscountEffectsFormset(EffectsFormset):
    model = BasketDiscountEffect
    def get_queryset(self):
        return self.owner.basket_discount_effects.all() # Is .all() needed?

class BasketLineEffectsFormset(EffectsFormset):
    model = BasketLineEffect
    def get_queryset(self):
        return self.owner.basket_line_effects.all() # Is .all() needed?
```